src/app/tabs/app-web3-provider.js

```diff
@@ -0,0 +1,31 @@
+import { Plugin } from '@remixproject/engine'
+import * as packageJson from '../../../package.json'
+
+export const profile = {
+  name: 'appWeb3Provider',
+  displayName: 'Global Web3 Provider',
+  description: 'Represent the current web3 provider used by the app at global scope',
+  methods: ['sendAsync'],
+  version: packageJson.version,
+  kind: 'provider'
+}
+
+export class AppWeb3Provider extends Plugin {
+  constructor (blockchain) {
+    super(profile)
+    this.blockchain = blockchain
+  }
+
+  /*
+    that is used by plugins to call the current ethereum provider.
+    Should be taken carefully and probably not be release as it is now.
+  */
+  sendAsync (payload) {
```

Comment:
Use JSDoc synthax to describe the method. Also, `payload` doesn't provider any information about the content. Consider precising the type in the `@params` documentation.